rules/windows/lateral_movement_via_wsus_update.toml

```diff
@@ -0,0 +1,71 @@
+[metadata]
+creation_date = "2024/07/19"
+integration = ["endpoint", "windows", "system","sentinel_one_cloud_funnel", "m365_defender"]
+maturity = "production"
+updated_date = "2024/07/19"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies a potential Windows Server Update Services (WSUS) abuse to execute attackers' payloads for lateral movement.
+WSUS is limited to executing Microsoft signed binaries, which limits the executables that can be used to tools published
+by Microsoft or Windows native executables.
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.process-*", "logs-windows.sysmon_operational-*", "endgame-*", "logs-sentinel_one_cloud_funnel.*", "logs-m365_defender.event-*", "logs-system.security-*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential WSUS Abuse for Lateral Movement"
+references = ["https://www.thehacker.recipes/a-d/movement/mitm-and-coerced-authentications/wsus-spoofing"]
+risk_score = 21
+rule_id = "8e2485b6-a74f-411b-bf7f-38b819f3a846"
+severity = "low"
+tags = [
+    "Domain: Endpoint",
+    "OS: Windows",
+    "Use Case: Threat Detection",
+    "Tactic: Lateral Movement",
+    "Data Source: Elastic Endgame",
+    "Data Source: Elastic Defend",
+    "Data Source: Sysmon",
+    "Data Source: SentinelOne",
+    "Data Source: Microsoft Defender for Endpoint"
+]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+process where host.os.type == "windows" and event.type == "start" and process.parent.name : "wuauclt.exe" and
+process.executable : "?:\\Windows\\SoftwareDistribution\\Download\\Install\\*" and
+(
+    process.name : (
+        "cmd.exe", "cscript.exe", "ieexec.exe", "iexpress.exe", "installutil.exe", "Microsoft.Workflow.Compiler.exe",
+        "msbuild.exe", "mshta.exe", "msiexec.exe", "msxsl.exe", "net.exe", "net1.exe", "powershell.exe", "pwsh.exe",
+        "reg.exe", "RegAsm.exe", "RegSvcs.exe", "regsvr32.exe", "rundll32.exe", "vssadmin.exe", "wbadmin.exe",
+        "wmic.exe", "wscript.exe", "bitsadmin.exe", "certutil.exe", "schtasks.exe", "cmstp.exe", "sc.exe", "curl.exe",
+        "ScriptRunner.exe", "Workfolders.exe", "psexec64.exe"
```

Comment:
Although the blogs document that this is abusable, I couldn't reproduce it with lolbins, and I suspect it was due to the way they were signed (using catalog instead of holding the signature in the file itself).

Reply:
limiting the rule to lolbins and running from SoftwareDistribution folder looks like a duplicate of this https://github.com/elastic/protections-artifacts/blob/c568d300cda114a25a95bfa96eec774b322b598e/behavior/rules/windows/defense_evasion_binary_masquerading_via_untrusted_path.toml#L9 (ms lolbins/shell script interpreter running from outside of syswow64/system32).  maybe just duplicate this endpoint rule (if its not already in SIEM) and adjust this wsus one to unsigned binaries ?

Reply:
WSUS can only run signed bins. I think the process original name part is important as the executable's name is controlled by the attacker, PSExec seems to be the one most commonly abused here, should we add it to that list?

Reply:
then a generic DR looking for lolbins in the list by process.pe.original_file_name or process.name and not running from system32 or syswow64 or microsoft.net folder will catch this scenario and other. IIRC we do have one in SIEM ?

Reply:
| PSExec seems to be the one most commonly abused here, should we add it to that list?

Psexec and unsigned and recently dropped could be a good candidate, rundll32.exe with DLL argument pointing to wsus folder ?

Reply:
> then a generic DR looking for lolbins in the list by process.pe.original_file_name or process.name and not running from system32 or syswow64 or microsoft.net folder will catch this scenario and other.

Sounds OK, but does it makes sense to keep them all in one rule? Sounds harder to tune and investigate

> IIRC we do have one in SIEM ?

I don't think so

> rundll32.exe with DLL argument pointing to wsus folder ?

Not sure if I'm following, can you elaborate?

Reply:
> if rundll32 is supplied as a bin/payload for sharpwsus then it will be copied to SoftwareDistribution ? if so then no need for a dedicated wsus rule unless its will be limited to psexec.

I couldn't reproduce with other binaries that don't contain the signature in the file itself (like psexec), although there are [mentions](https://www.thehacker.recipes/a-d/movement/mitm-and-coerced-authentications/wsus-spoofing) that this is abusable with lolbins:

> Programs other than PsExec.exe can be used here. Using built-in programs features to bypass security restrictions or operate attacks like this is called [Living off the land](https://www.thehacker.recipes/infra/privilege-escalation/windows/living-off-the-land) (LOL). Other Windows LOL binaries and scripts (a.k.a. LOLbins or LOLbas) can be found on [lolbas-project.github.io](https://lolbas-project.github.io/#).

So, should I restrict this one to psexec and duplicate the ER one for the data sources other than Endpoint?

Reply:
| So, should I restrict this one to psexec and duplicate the ER one for the data sources other than Endpoint?
sounds like a good plan

Reply:
Done, will put the conversion of the ER in my backlog